apps/tx-builder/src/components/Builder.tsx

```diff
@@ -0,0 +1,319 @@
+  && {
+    width: 520px;
+    margin-bottom: 10px;
+  }
+`;
+
+const StyledSelect = styled(Select)`
+  width: 520px;
+`;
+
+const StyledExamples = styled.div`
+  button {
+    padding: 0;
+  }
+`;
+
+type Props = {
+  contract: ContractInterface | undefined;
+  to: string | null;
+};
+
+export const Builder = ({ contract, to }: Props): ReactElement | null => {
+  const { sdk, safe } = useSafeAppsSDK();
+  const services = useServices(safe.network);
+
+  const [toInput, setToInput] = useState('');
+  const [valueInput, setValueInput] = useState('');
+
+  const [transactions, setTransactions] = useState<ProposedTransaction[]>([]);
+  const [reviewing, setReviewing] = useState(false);
+  const [selectedMethodIndex, setSelectedMethodIndex] = useState(0);
+  const [showExamples, setShowExamples] = useState(false);
+  const [addTxError, setAddTxError] = useState<string | undefined>();
+  const [inputCache, setInputCache] = useState<string[]>([]);
+  const [isValueInputVisible, setIsValueInputVisible] = useState(false);
+
+  const handleMethod = useCallback(
+    async (methodIndex: number) => {
+      if (!contract || contract.methods.length <= methodIndex) return;
+      setSelectedMethodIndex(methodIndex);
+    },
+    [contract],
+  );
+
+  const handleInput = useCallback(
+    async (inputIndex: number, input: string) => {
+      inputCache[inputIndex] = input;
+      setInputCache(inputCache.slice());
+    },
+    [inputCache],
+  );
+
+  const getInputHelper = (input: any) => {
+    // This code renders a helper for the input text.
+    if (input.type.startsWith('tuple')) {
+      return `tuple(${input.components.map((c: any) => c.internalType).toString()})${
+        input.type.endsWith('[]') ? '[]' : ''
+      }`;
+    } else {
+      return input.type;
+    }
+  };
+
+  const getContractMethod = useCallback(() => contract?.methods[selectedMethodIndex], [contract, selectedMethodIndex]);
+
+  const handleSubmit = () => {
+    sendTransactions();
+    setTransactions([]);
+    setReviewing(false);
+  };
+
+  const handleDismiss = () => {
+    setReviewing(false);
+  };
+
+  const addTransaction = async () => {
+    let description = '';
+    let data = '';
+    const web3 = services.web3;
+
+    if (!web3) {
+      return;
+    }
+
+    if (contract && contract.methods.length > selectedMethodIndex) {
+      const method = contract.methods[selectedMethodIndex];
+      const cleanInputs = [];
+      description = method.name + ' (';
+      for (let i = 0; i < method.inputs.length; i++) {
+        const cleanValue = inputCache[i] || '';
+        cleanInputs[i] = cleanValue.charAt(0) === '[' ? JSON.parse(cleanValue.replace(/"/g, '"')) : cleanValue;
+        if (i > 0) {
+          description += ', ';
+        }
+        const input = method.inputs[i];
+        description += (input.name || input.type) + ': ' + cleanValue;
+      }
+      description += ')';
+
+      try {
+        if (!['receive', 'fallback'].includes(method.name)) {
```

Comment:
Should we skip the whole description/ method parsing (so have this check at the beginning of the if in line 113) and directly go the the next part?